# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### Chosen Lane

**Refresh / Content Opportunity Scoring**

### Why I chose this lane

I chose the **Refresh / Content Opportunity Scoring** lane because it focuses on identifying webpages that should be reviewed first for updates based on search performance and engagement metrics. The starter dataset already contains search, traffic, CTR, engagement, and freshness information, making it suitable for building evidence-based recommendations. This project aims to help content teams prioritize pages that deserve review instead of relying only on manual analysis.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Research Question
Which webpages should be prioritized for content refresh based on search performance and engagement metrics?

### Decision
Determine which webpages should be reviewed first for content improvement.

### Action
The SEO or content team will update, rewrite, expand, or monitor the selected webpages.

### Unit of Analysis
One webpage (one content item).

### Cost of a Wrong Recommendation
If the recommendation is incorrect, the team may spend time updating pages that do not require improvement while overlooking pages that actually need attention. This results in wasted effort, reduced productivity, and inefficient use of SEO resources.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

# Load the dataset
df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

# Display basic information
print("Dataset Loaded Successfully!")
print("Shape:", df.shape)

display(df.head())

Dataset Loaded Successfully!
Shape: (30000, 44)


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [2]:
print("="*50)
print("DATASET OVERVIEW")
print("="*50)

print(f"Total Rows    : {df.shape[0]}")
print(f"Total Columns : {df.shape[1]}")

print("\nColumn Names:")
print(df.columns.tolist())

print("\nData Types:")
display(df.dtypes)

print("\nMissing Values:")
display(df.isnull().sum())

DATASET OVERVIEW
Total Rows    : 30000
Total Columns : 44

Column Names:
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']

Data Types:


content_id                    str
client_id                     str
search_volume             float64
competition               float64
competition_level             str
cpc                       float64
content_type                  str
main_intent                   str
word_count                float64
char_count                float64
provider_used                 str
model_used                    str
impressions_90d             int64
clicks_90d                  int64
pageviews_90d               int64
sessions_90d                int64
users_90d                   int64
engaged_sessions_90d        int64
ai_sessions_90d             int64
scroll_events_90d           int64
days_with_impressions       int64
days_with_sessions          int64
impressions_last_30d        int64
clicks_last_30d             int64
sessions_last_30d           int64
impressions_prev_30d        int64
clicks_prev_30d             int64
sessions_prev_30d           int64
content_age_days            int64
age_tier      


Missing Values:


content_id                    0
client_id                     0
search_volume              2468
competition                2468
competition_level          2610
cpc                        2468
content_type                  0
main_intent                2374
word_count                 7699
char_count                 7699
provider_used             21438
model_used                 5733
impressions_90d               0
clicks_90d                    0
pageviews_90d                 0
sessions_90d                  0
users_90d                     0
engaged_sessions_90d          0
ai_sessions_90d               0
scroll_events_90d             0
days_with_impressions         0
days_with_sessions            0
impressions_last_30d          0
clicks_last_30d               0
sessions_last_30d             0
impressions_prev_30d          0
clicks_prev_30d               0
sessions_prev_30d             0
content_age_days              0
age_tier                      0
age_tier_order                0
days_sin

In [3]:
print("="*50)
print("KEY STATISTICS")
print("="*50)

print(f"Total Pages: {len(df)}")
print(f"Total Features: {df.shape[1]}")

print(f"Average Impressions (90d): {df['impressions_90d'].mean():.2f}")
print(f"Average Sessions (90d): {df['sessions_90d'].mean():.2f}")
print(f"Average Position: {df['avg_position'].mean():.2f}")

KEY STATISTICS
Total Pages: 30000
Total Features: 44
Average Impressions (90d): 5200.37
Average Sessions (90d): 37.07
Average Position: 16.34


### Quick Look at the Data

The starter dataset contains **30,000 webpages** and **44 features** related to search performance, engagement, content quality, and page freshness.

Some important observations from the dataset include:

- **Total Pages:** 30,000
- **Total Features:** 44
- **Average 90-day Impressions:** 5200.37
- **Average 90-day Sessions:** 37.07
- **Average Google Search Position:** 16.34

These statistics indicate that the dataset contains sufficient historical search and engagement information to support the **Refresh / Content Opportunity Scoring** lane. The available metrics will help identify pages that should be prioritized for review and content updates.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What I Can Claim
- I can identify webpages that appear to be good candidates for content refresh using historical search and engagement metrics.
- I can rank webpages based on observable search performance and engagement signals.
- I can provide decision support to help prioritize which pages should be reviewed first.

### What I Cannot Claim
- I cannot guarantee that refreshing a webpage will improve its Google ranking.
- I cannot prove that a content update directly causes an increase in traffic or rankings.
- My analysis is based on historical observations and should be used to support, not replace, human decision-making.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.